src/plugins/intel_cpu/src/edge.cpp

```diff
@@ -65,22 +65,39 @@ void Edge::drop() {
     _drop_from(getChild()->parentEdges);
 }
 
-void Edge::collectConsumers(std::vector<NodePtr>& result) const {
+void Edge::collectConsumers(std::vector<NodePtr>& result, bool nested) const {
     if (!this->getChild()->getChildEdges().empty() && this->inPlace(LOOK_DOWN)) {
         if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
             auto peerOutputNum = this->getOutputNum();
             auto peerInPlacePort = peerChildSPD->getConfig().inConfs[peerOutputNum].inPlace();
             auto& vecChildEdges = this->getChild()->getChildEdgesAtPort(peerInPlacePort);
             for (auto childEdge : vecChildEdges) {
-                childEdge->collectConsumers(result);
+                childEdge->collectConsumers(result, true);
             }
         }
-    } else {
-        auto childNode = this->getChild();
-        if (Type::ShapeOf == childNode->getType()) {
-            // ShapeOf doesn't actually read the data, it only reads shape
-            return;
+        return;
+    } else if (!this->getChild()->getChildEdges().empty()) {
+        if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
+            bool isBaseEdge = false;
+            for (size_t i = 0; i < peerChildSPD->getConfig().outConfs.size(); i++) {
+                auto peerOutPlacePort = peerChildSPD->getConfig().outConfs[i].inPlace();
+                if (peerOutPlacePort == this->getOutputNum()) {
+                    isBaseEdge = true;
+                    for (auto childEdge : this->getChild()->getChildEdgesAtPort(i))
+                        childEdge->collectConsumers(result, true);
+                }
+            }
+            if (isBaseEdge)
+                return;
         }
+    }
+
+    auto childNode = this->getChild();
+    if (Type::ShapeOf == childNode->getType()) {
+        // ShapeOf doesn't actually read the data, it only reads shape
+        if (nested)
+            result.push_back(this->getParent());
```

Comment:
Think about this case -

-> transpose ---> reshape (inplace up) -> MVN (inplace up) -> shapeof --> ...
                       |--> reshape (inplace) -> subgraph

any problem here? L98 the nested check and L90 isBaseEdge check will block pushing any consumer into result vector.

Reply:
Actually it comes from this PR https://github.com/openvinotoolkit/openvino/pull/21372/files . I guess shapeof doesn't read the content of memory and its child isn't a true consumer, so just return the parent of shapeof as the consumer.

Reply:
With this case, there is no consumer returned, is it?

Reply:
in this case, the consumer of edge: transpose->reshape(inplace up) is MVN. While the consumer of edge: MVN->shapeof is nothing.